### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by 0sugo**
In this PR I did the following:

- Created a class Classroom with the following:
    - @label instance variable, should be initialized in the constructor.
    - Setter and getter for @label (remember about attr_accessor).
- Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
    - Created the has-many side (a classroom has many students).
    - Created the belongs-to side (a student belongs to a classroom).
    - Made sure that when adding a student to a classroom it also sets the classroom for the student.
    - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class Book with the following:
    - @title and @author instance variables, should be initialized in the constructor.
    - Setters and getters for instance variables (remember about attr_accessor).
- Created a class Rental with the following:
    - @date instance variable, should be initialized in the constructor.
    - Setter and getter for @date (remember about attr_accessor).
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
    - Created the has-many side of Book and Rental (a book has many rentals).
    - Created the belongs-to side of Rental and Book (a rental belongs to a book).
    - Created the has-many side of Person and Rental (a person has many rentals).
    - Created the belongs-to side of Rental and Person (a rental belongs to a person).
    - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 # Define student
 class Student < Person
-  def initialize(classroom)
+  attr_accessor :name, :classroom
+
+  def initialize(_classroom)
     super()
-    @classroom = classroom
+    @name = name
+    @classroom = nil
```
Comment (tufoinnkuo10) [inline student.rb]: - [x] Great job linking the classes. please kindly add a  method that serves as the 'belongs-to' side (a student belongs to a classroom). you can get more light to that [here](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`)
eg
```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: person.rb [hunk 2]
```diff
@@ -10,9 +16,14 @@ def initialize(age = nil, name = 'unknown', parent_permission: true)
...
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(book, date)
+    Rental.new(date, book, self)
   end
 
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
```
Comment (tufoinnkuo10) [inline person.rb]: - [optional] please kindly take your attr_accessor up before the beginning of your methods. As you know, it helps tell Ruby to create your methods for you. I.e. it creates both the READER & WRITER methods

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @0sugo 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ All classes available
✅ associations ok

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: student.rb (hunk 1 above)
Comment (0sugo) [reply student.rb]: Thanks for the keen observation @tufoinnkuo10  I've done that.

File: person.rb (hunk 2 above)
Comment (0sugo) [reply person.rb]: I've attended to that as well

**Review by sumairq — APPROVED** [review]
### Hi @0sugo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**